### ISSUE TITLE
Some questions about Wayland

### ISSUE THREAD

**Issue by oznakn** [Askannz/optimus-manager#199]
Hi,

First of all, thanks for the great work. There are some solutions on the internet for the nvidia-intel laptops, but not like Optimus Manager. None of them solves my problems. 

I am wondering about is there a particular reason that Optimus Manager has no Wayland support? Is it just the Optimus Manager is not designed to be work on Wayland and can be ported, or there are limitations on Wayland that blocks Optimus Manager to be work with it.

Have a nice day!

**Comment by Askannz** [comment]
Hi, I'll try to answer your question.

I'm not very familiar with Wayland, but as far as I understand, there is no standard way to make the Wayland compositor choose a particular GPU as the "primary" one. Sway has the `WLR_DRM_DEVICES` env variable and other compositors may have something similar, but it seems to be left to the implementation. So unlike Xorg, we can't just auto-generate some config file that points to the Nvidia GPU, and have it work for every desktop environment.

But even if we could, Wayland is still a poor experience on Nvidia anyways. Gnome has a somewhat working implementation (but still they prefer disabling it by default), KDE only recently started supporting it, XFCE doesn't support it at all, and Sway has such a [hate boner](https://github.com/swaywm/sway/issues/3039) for the proprietary driver that it will refuse to start (with a condescending error message) if it detects the `nvidia` module, even if you are manually selecting the Intel card with `WLR_DRM_DEVICES`.

Worse, most applications still rely on Xorg to access the GPU, but they can't do that in Wayland because Nvidia doesn't support Xwayland. So even if you can get a Wayland compositor running on the Nvidia card, there is really no point to do it since virtually no application will be able to take advantage of your powerful GPU anyways. And that's not even mentioning the many issues that games have with XWayland, and are not specific to Nvidia.

So in short, there's really no incentive to add Wayland support to optimus-manager at the moment.

But from my (arguably limited) knowledge about the Linux graphics stack, Wayland still seems like the proper way to go for the Linux desktop, so I'm hoping the situation will be less dire a few years from now.

**Comment by oznakn** [comment]
Hi,

Thanks for this explanatory answer. I hope in the end, all Nvidia related issues on Linux machines will be solved. It is really painful for most people.

**Comment by Terryvanderjagt** [comment]
Hi, 

I would like to add a question about wayland. I rarely use my Nvidia GPU and mostly use my integrated card and I prefer to use wayland by default. I use GNOME+GDM and I understand from the Readme that I need to disable wayland for GDM. Is it possible however to start a wayland session when I am using the integrated Intel CPU?

Thanks in advance,
Terry

**Comment by KingTheGuy** [comment]
> Hi,
> 
> I would like to add a question about wayland. I rarely use my Nvidia GPU and mostly use my integrated card and I prefer to use wayland by default. I use GNOME+GDM and I understand from the Readme that I need to disable wayland for GDM. Is it possible however to start a wayland session when I am using the integrated Intel CPU?
> 
> Thanks in advance,
> Terry

I'm on gnome and what i do is after I've installed and setup optimus-manager i then set the desktop to wayland and create another grub entry that uses optimus-manager.startup=nvidia or hybrid this pretty much forces it back to xorg whenever i boot into that one. when im using my pc as a desktop i boot with the startup parameter, but when im out and about i just boot normally which again uses wayland. anyways, does the job pretty well.
